kong/router.lua

```diff
@@ -748,12 +748,15 @@ local function sort_routes(r1, r2)
     end
   end
 
-  do
-    local rp1 = r1.route.regex_priority or 0
-    local rp2 = r2.route.regex_priority or 0
+  -- only regex path use regex_priority 
+  if band(r1.submatch_weight,MATCH_SUBRULES.HAS_REGEX_URI) ~= 0 then
```

Comment:
this doesn't sound right to me.  rule of "if no regex, then ignore regex_priority" should be applied independently.

Reply:
I add a unit test,  maybe it can explain what I say.
this test will failed on master.
the problem is a route  with path=/.*   host=domain1,  it can affect other domain's route.

Reply:
I see now.  not about the test, but I was worried that this `HAS_REGEX_URI` condition on `r1` didn't cover all possible combinations, but now I realize that this only happens when the `.submatch_weight` fields are equal, so testing on a single one is appropriate.